src/components/Pagination/Pagination.js

```diff
@@ -0,0 +1,164 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import bem from 'bem';
+import styles from './Pagination.scss';
+import Button from '../Button';
+import PaginationButton from './PaginationButton';
+
+const { block, elem } = bem({
+    name: 'Pagination',
+    classnames: styles,
+    propsToMods: ['align']
+});
+
+const Pagination = props => {
+    const {
+        align,
+        currentPage,
+        maxPageButtons,
+        totalPages,
+        prevLabel,
+        nextLabel,
+        lastLabel,
+        firstLabel,
+        onClick,
+        ...rest
+    } = props;
+
+    const defineRange = () => {
+        const showBefore = Math.floor((maxPageButtons - 1) / 2);
+        const start = Math.max(
+            2,
+            Math.min(currentPage - showBefore, totalPages - maxPageButtons + 1)
+        );
+        const end = Math.min(totalPages, start + maxPageButtons - 2);
+        const range = new Array(end - start + 1).fill().map((_, i) => start + i);
+
+        return range;
+    };
+
+    const isPrevDisabled = currentPage === 1;
+    const isNextDisabled = currentPage === totalPages;
```

Comment:
`next` or `last` will not be disabled if somehow `currentPage > totalPages`. Since neither variable is owned by this component, it'd make sense to at least not provide buttons to non-existent pages.

Same goes for `currentPage < 1`.

`currentPage={100} totalPages={8} maxPageButtons={5}`
`currentPage={0} totalPages={8} maxPageButtons={5}`

Reply:
now it throws an error if  current page is illogical